docs/README.md

```diff
@@ -0,0 +1,287 @@
+    * [Authorization code grant (interactive clients)](https://docs.microsoft.com/en-us/rest/api/azure/#authorization-code-grant-interactive-clients)
+    * [Client credentials grant (non-interactive clients)](https://docs.microsoft.com/en-us/rest/api/azure/#client-credentials-grant-non-interactive-clients)
+  * [Assemble the request message](https://docs.microsoft.com/en-us/rest/api/azure/#assemble-the-request-message)
+    * [Request URI](https://docs.microsoft.com/en-us/rest/api/azure/#request-uri)
+    * [Request header](https://docs.microsoft.com/en-us/rest/api/azure/#request-header)
+    * [Request body](https://docs.microsoft.com/en-us/rest/api/azure/#request-body)
+* [Send the request](https://docs.microsoft.com/en-us/rest/api/azure/#send-the-request)
+* [Process the response message](https://docs.microsoft.com/en-us/rest/api/azure/#process-the-response-message)
+* [Async operations, throttling, and paging](https://docs.microsoft.com/en-us/rest/api/azure/#async-operations-throttling-and-paging)
+
+# Using ms_rest_azure client library
+
+The ms_rest_azure is a rest api client library out of box in Ruby SDK. Since it will be deprecated together, it is recommanded that you implement your own client according to your needs. The following documents are intend to help you to understand the concepts and workflow by using it.
+
+## Install
+
+You can install the ms_rest_azure gem with the following command:
+
+```ruby
+gem install 'ms_rest_azure'
+```
+
+## Prerequisite
+
+In order to use the Azure client, you must supply the following values:
+
+* Tenant Id
+* Client Id
+* Subscription Id
+* Client Secret
+
+You could pass the above values in the following ways:
+
+### Option 1 - Environment Variables
+You can set the (above) values using the following environment variables:
+
+* AZURE_TENANT_ID
+* AZURE_CLIENT_ID
+* AZURE_SUBSCRIPTION_ID
+* AZURE_CLIENT_SECRET
+
+To set the environment variables, in Windows, you could use the command such as:
+
+```
+set AZURE_TENANT_ID=<YOUR_TENANT_ID>
+```
+
+In Unix based systems, you could use the command such as:
+
+```
+export AZURE_TENANT_ID=<YOUR_TENANT_ID>
+```
+
+### Option 2 - Options Hash
+In addition to environment variables, you can also pass those parameters in an Option Hash when you initialize the profile clients. This options hash consists of tenant_id, client_id, client_secret, subscription_id, active_directory_settings and credentials. Among these, the active_directory_settings and credentials are optional.
+
+You can set the (above) values using the options hash:
+
+```ruby
+options = {
+  tenant_id: 'YOUR TENANT ID',
+  client_id: 'YOUR CLIENT ID',
+  client_secret: 'YOUR CLIENT SECRET',
+  subscription_id: 'YOUR SUBSCRIPTION ID'
+}
+```
+
+## OAuth2 Token Credentials
+
+[Acquire an acces token](https://docs.microsoft.com/en-us/rest/api/azure/#acquire-an-access-token)
+
+### Use token provider
```

Comment:
are token providers mandatory in our scenario?  if so, we should make it more clear like Step 1 -> Step 2 -> Step 3